resources/assets/coffee/react/profile-page/info.coffee

```diff
@@ -35,64 +35,61 @@ class ProfilePage.Info extends React.Component
+          el 'p', className: 'profile-info__title profile-info__title--supporter',
             Lang.get('users.show.is_supporter')
 
         if @props.user.title?
-          el 'p', className: 'profile-title', @props.user.title
+          el 'p', className: 'profile-info__title', @props.user.title
 
-      el 'div', className: 'compact profile-row',
+      el 'div', className: 'page-contents__row',
         if @originKeys().length
-          el 'p', null,
+          el 'p', className: 'profile-info__location', null,
             Lang.get "users.show.origin.#{@originKeys().join('_')}",
               country: @props.user.country.name
               age: @props.user.age
 
         if @props.user.location
-          el 'p', null,
+          el 'p', className: 'profile-info__location', null,
             Lang.get 'users.show.current_location', location: @props.user.location
 
       el 'p',
-        className: 'profile-row'
-        dangerouslySetInnerHTML:
-          __html: Lang.get 'users.show.lastvisit', date: osu.timeago(@props.user.lastvisit)
-
-      if @props.user.twitter
-        el 'dl', className: 'profile-data profile-row',
-          el 'dt', null, 'Twitter'
-          el 'dd', null,
-            el 'a', href: "https://twitter.com/#{@props.user.twitter}",
-              "@#{@props.user.twitter}"
-
-      if @props.user.skype
-        el 'dl', className: 'profile-data profile-row',
-          el 'dt', null, 'Skype'
-          el 'dd', null,
-            el 'a', href: "skype:#{@props.user.skype}?chat", @props.user.skype
-
-      if @props.user.lastfm
-        el 'dl', className: 'profile-data profile-row',
-          el 'dt', null, 'Last.fm'
-          el 'dd', null,
-            el 'a', href: "https://last.fm/user/#{@props.user.lastfm}",
-              @props.user.lastfm
-
-      if @props.user.playstyle.length
-        el 'dl', className: 'profile-data profile-row',
-          el 'dt', null, Lang.get('users.show.plays_with._')
-          el 'dd', null,
-            @props.user.playstyle.map (s) ->
-              Lang.get "users.show.plays_with.#{s}"
-            .join ', '
+        className: 'page-contents__row'
+        dangerouslySetInnerHTML: { __html: Lang.get 'users.show.lastvisit', date: osu.timeago(@props.user.lastvisit) }
+
+      datas.map (m) =>
+        switch m
+          when 'twitter'
+            return unless @props.user.twitter
+            dt = 'Twitter'
+            dd = el 'a', href: "https://twitter.com/#{@props.user.twitter}", "@#{@props.user.twitter}"
+          when 'skype'
+            return unless @props.user.skype
+            dt = 'Skype'
+            dd = el 'a', href: "skype:#{@props.user.skype}?chat", @props.user.skype
+          when 'lastfm'
+            return unless @props.user.lastfm
+            dt = 'Last.fm'
+            dd = el 'a', href: "https://last.fm/user/#{@props.user.lastfm}", @props.user.lastfm
+          when 'playstyles'
+            return unless @props.user.playstyle.length
```

Comment:
updating the getter method in the model to return `null` when it's empty will simplify check for this part (`return if !@props.user[m]?`